providers/httpnet/api.go

```diff
@@ -0,0 +1,260 @@
+package httpnet
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/StackExchange/dnscontrol/v3/pkg/diff"
+	"golang.org/x/net/idna"
+)
+
+const endpoint = "https://partner.routing.net/api/%s/v1/json/%s"
+
+type api struct {
```

Comment:
Please use httpnetProvider for this struct.

You can rename this automatically with:

    gorename -v -from '"github.com/StackExchange/dnscontrol/v3/providers/httpnet".api' -to 'httpnetProvider'